src/rpc/ngHandlers/AccountChannels.h

```diff
@@ -0,0 +1,130 @@
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#pragma once
+
+#include <backend/BackendInterface.h>
+#include <rpc/common/Types.h>
+#include <rpc/common/Validators.h>
+
+#include <boost/asio/spawn.hpp>
+
+#include <functional>
+#include <vector>
+
+namespace RPCng {
+class AccountChannelsHandler
+{
+public:
+    // type align with SField.h
+    struct ChannelResponse
+    {
+        std::string channelID;
+        std::string account;
+        std::string accountDestination;
+        std::string amount;
+        std::string balance;
+        std::optional<std::string> publicKey;
+        std::optional<std::string> publicKeyHex;
+        uint32_t settleDelay;
+        std::optional<uint32_t> expiration;
+        std::optional<uint32_t> cancelAfter;
+        std::optional<uint32_t> sourceTag;
+        std::optional<uint32_t> destinationTag;
+    };
+
+    struct HandlerOutput
+    {
+        std::vector<ChannelResponse> channels;
+        std::string account;
+        std::string ledgerHash;
+        uint32_t ledgerIndex;
+        // validated should be sent via framework
+        bool validated = true;
+        uint32_t limit;
+        std::optional<std::string> marker;
+    };
+
+    struct HandlerInput
+    {
+        std::string account;
+        std::optional<std::string> destinationAccount;
+        std::optional<std::string> ledgerHash;
+        std::optional<uint32_t> ledgerIndex;
+        uint32_t limit = 50;
+        std::optional<std::string> marker;
+    };
+
+    using Input = HandlerInput;
+    using Output = HandlerOutput;
+    using Result = RPCng::HandlerReturnType<Output>;
+
+    AccountChannelsHandler(
+        boost::asio::yield_context& yieldCtx,
+        std::shared_ptr<BackendInterface>& sharedPtrBackend)
+        : yieldCtx_(yieldCtx), sharedPtrBackend_(sharedPtrBackend)
+    {
+    }
+
+    RpcSpecConstRef
+    spec() const
+    {
+        // clang-format off
+        static const RpcSpec rpcSpec = {
+            {"account", validation::Required{}, validation::AccountValidator},
+            {"destination_account", validation::Type<std::string>{},validation::AccountValidator},
+            {"ledger_hash", validation::LedgerHashValidator},
+            {"limit", validation::Type<uint32_t>{},validation::Between{10,400}},
+            {"ledger_index", validation::LedgerIndexValidator},
+            {"marker", validation::MarkerValidator}
+        };
+        // clang-format on
+
+        return rpcSpec;
+    }
+
+    Result
+    process(Input input) const;
+
+private:
+    // dependencies
+    const std::reference_wrapper<boost::asio::yield_context> yieldCtx_;
```

Comment:
Sadly we can't have this as a dependency since it's a new context for each new rpc call. I think we can introduce some artificial `asio::spawn` inside of our `RPCHelpers` just to avoid having this dependency here. Eventually we should be able to completely remove the need for these `yield_context`s that leak everywhere throughout `clio` atm. 😢

Reply:
I think the yield is from our work queue. Previously, context will hold the reference of it and yield when db operation happens. If we want to artifically 'spawn' something, where we should execute it ? another thread pool?